src/main/java/org/schemaspy/service/DatabaseService.java

```diff
@@ -84,42 +84,31 @@ public void gatheringSchemaDetails(Config config, Database db, ProgressListener
    private void initCatalogs(Config config, Database db, ProgressListener listener) throws SQLException {
 
             String sql = Config.getInstance().getDbProperties().getProperty("selectCatalogsSql");
-            PreparedStatement stmt = null;
-			ResultSet rs =  null;
+
             if (sql != null && db.getCatalog() != null) {
-                try {
-                stmt = sqlService.prepareStatement(sql,db,null);
-                rs = stmt.executeQuery();
-                    while (rs.next()) {
+                try (PreparedStatement stmt = sqlService.prepareStatement(sql, db, null);
+                     ResultSet rs = stmt.executeQuery();) {
+                    if (rs.next()) {
                          db.getCatalog().setComment(rs.getString("catalog_comment"));
-                         break;
                     }
                 } catch (SQLException sqlException) {
                     //db.getSchema().setComment(null);
-                } finally {// 
-                    stmt.close();
-                  	rs.close();
                 }
             }
     }
 
     private void initSchemas(Config config, Database db, ProgressListener listener) throws SQLException {
     	  String sql = Config.getInstance().getDbProperties().getProperty("selectSchemasSql");
-          PreparedStatement stmt = null;
-			ResultSet rs =  null;
+
           if (sql != null &&  db.getSchema() != null) {
-              try {
-              stmt = sqlService.prepareStatement(sql,db,null);
-              rs = stmt.executeQuery();
-                  while (rs.next()) {
+              try (PreparedStatement stmt = sqlService.prepareStatement(sql, db, null);
+                   ResultSet rs = stmt.executeQuery()) {
+
+                  if (rs.next()) {
                        db.getSchema().setComment(rs.getString("schema_comment"));
-                       break;
                   }
               } catch (SQLException sqlException) {
                   //db.getSchema().setComment(null);
```

Comment:
The same situation is here, I know that it was before but it will be good to not leave in the code empty `catch`